Tasks/ServiceFabricDeploy/deploy.ps1

```diff
@@ -0,0 +1,84 @@
+try {
+    # Import the localized strings. 
+    Import-VstsLocStrings "$PSScriptRoot\task.json"
+    
+    # Load utility functions
+    . "$PSScriptRoot\utilities.ps1"
+    
+    # Collect input values
+    $publishProfilePathSearchPattern = Get-VstsInput -Name publishProfilePath -Require
+    Write-Host (Get-VstsLocString -Key SearchingForPublishProfile -ArgumentList $publishProfilePathSearchPattern) 
+    $publishProfilePath = Find-VstsFiles -LegacyPattern $publishProfilePathSearchPattern
+    Assert-SingleItem $publishProfilePath $publishProfilePathSearchPattern
+    Assert-VstsPath -LiteralPath $publishProfilePath -PathType Leaf
+    Write-Host (Get-VstsLocString -Key FoundPublishProfile -ArgumentList $publishProfilePath)
+    
+    $applicationPackagePathSearchPattern = Get-VstsInput -Name applicationPackagePath -Require
+    Write-Host (Get-VstsLocString -Key SearchingForApplicationPackage -ArgumentList $applicationPackagePathSearchPattern)
+
+    # Workaround for bug in Find-VstsFiles that doesn't handle the IncludeDirectories parameter correctly. Modifies the search pattern to look for app manifest file instead of the directory
+    # and then get the directory of that file.
+    $applicationPackagePathSearchPattern = Join-Path $applicationPackagePathSearchPattern "ApplicationManifest.xml"
+    $applicationPackagePath = Find-VstsFiles -LegacyPattern $applicationPackagePathSearchPattern
+    Assert-SingleItem $applicationPackagePath $applicationPackagePathSearchPattern
+    $applicationPackagePath = [System.IO.Path]::GetDirectoryName($applicationPackagePath)
+    Assert-VstsPath -LiteralPath $applicationPackagePath -PathType Container
+    Write-Host (Get-VstsLocString -Key FoundApplicationPackage -ArgumentList $applicationPackagePath)
+
+    $aadServiceConnectionName = Get-VstsInput -Name aadServiceConnectionName -Require
+    $aadConnectedServiceEndpoint = Get-VstsEndpoint -Name $aadServiceConnectionName -Require
+
+    $publishProfile = Read-PublishProfile $publishProfilePath
+    $clusterConnectionParameters = $publishProfile.ClusterConnectionParameters
+    
+    if (-not $clusterConnectionParameters.ContainsKey("AzureActiveDirectory") -or $clusterConnectionParameters.Item("AzureActiveDirectory") -eq $false)
+    {
+        throw (Get-VstsLocString -Key AzureActiveDirectoryNotSet)
+    }
+    
+    # Connect to cluster
+    $securityToken = Get-AadSecurityToken -ClusterConnectionParameters $clusterConnectionParameters -ConnectedServiceEndpoint $aadConnectedServiceEndpoint
+    $clusterConnectionParameters["SecurityToken"] = $securityToken
+    $clusterConnectionParameters["WarningAction"] = "SilentlyContinue"
+    [void](Connect-ServiceFabricCluster @clusterConnectionParameters)
+    Write-Host (Get-VstsLocString -Key ConnectedToCluster)
+    
+    $regKey = "HKLM:\SOFTWARE\Microsoft\Service Fabric SDK"
+    $moduleFolderPath = (Get-ItemProperty -Path $regKey -Name FabricSDKPSModulePath).FabricSDKPSModulePath
```

Comment:
It looks like the end user just ends up with a reg key not found error if this isn't installed. It would be better to fail with an actionable error message.